addons/payment_adyen/models/payment_provider.py

```diff
@@ -52,14 +49,15 @@ def write(self, values):
 
     @api.model
     def _adyen_trim_api_urls(self, values):
-        """ Remove the version and the endpoint from the url of Adyen API fields.
-
+        """ Remove everything exept prefix from the url of Adyen API fields.
         :param dict values: The create or write values
         :return: None
         """
-        for field_name in ('adyen_checkout_api_url', 'adyen_recurring_api_url'):
-            if values.get(field_name):  # Test the value in case we're duplicating a provider
-                values[field_name] = re.sub(r'[vV]\d+(/.*)?', '', values[field_name])
+
+        if values.get('adyen_api_url_prefix'):  # Test the value in case we're duplicating
+                                                # a provider if prefix was changed
```

Comment:
```suggestion
        if values.get('adyen_api_url_prefix'):  # Test and trim the value when needed
```
Not strictly necessary, so as you prefer =]